pkg/provider/provider.go

```diff
@@ -181,11 +177,13 @@ func (b bigAnimalProvider) DataSources(ctx context.Context) []func() datasource.
 	return []func() datasource.DataSource{
 		NewProjectsDataSource,
 		NewPgdDataSource,
+		NewRegionDataSource,
```

Comment:
Like the `NewProjectsDataSource` we need to rename this resource as `NewRegionsDataSource` as the data source is returning a list of regions, instead of a single region output. See #131

Reply:
ok, I will follow this design

Reply:
That's actually not enough :) 
We need to rename the data source from `biganimal_region` to `biganimal_regions`, update the examples and the documentation, and also mark the `biganimal_region` data source as deprecated. 

I wonder if we should do all those in this PR, or create a new one. I'm fine with both options.